src/Samples/SubModelSeq/Program.fs

```diff
@@ -13,177 +31,199 @@ module List =
         | a when a = j -> i
         | a -> a)
 
+  let swapWithNext i = swap i (i + 1)
+  let swapWithPrev i = swap i (i - 1)
+
+  let cons a ma = a :: ma
 
+        
 [<AutoOpen>]
-module Domain =
+module IdentifiableTypes =
+
+  type Identifiable<'a> =
+    { Id: Guid
+      Value: 'a }
+  
+  module Identifiable =
+    module Id =
+      let get m = m.Id
+      let set v m = { m with Id = v }
+      let map x = x |> map get set
+    module Value =
+      let get m = m.Value
+      let set v m = { m with Value = v }
+      let map f = f |> map get set
+      let mapIf p f = f |> map |> mapIf p
+      let mapIfId id f = f |> mapIf (fun m -> m.Id = id)
```

Comment:
Going by the built-in `Result.map` and `Result.mapError` (as opposed to `Result.Ok.map` and `Result.Error.map`), I suggest removing the inner `Id` and `Value` modules and renaming `Id.get` to `getId`. The ID `set` and `map` can be removed since they are unused (this is a sample, not a library). The value function names are fine.

```suggestion
  module Identifiable =
  
    let getId m = m.Id
    
    let get m = m.Value
    
    let set v m = { m with Value = v }
    
    let map f = f |> map get set
    
    let mapIf p f = f |> map |> mapIf p
    
    let mapIfId id f = f |> mapIf (fun m -> m.Id = id)
```

Reply:
Sure